pennylane_lightning/src/tests/CreateAllWires.hpp

```diff
@@ -0,0 +1,92 @@
+#pragma once
+#include "BitUtil.hpp"
+#include "Constant.hpp"
+#include "ConstantUtil.hpp"
+#include "GateOperation.hpp"
+
+#include <cstdlib>
+#include <vector>
+
+namespace Pennylane {
+
+class WiresGenerator {
+  public:
+    [[nodiscard]] virtual auto all_perms() const
+        -> const std::vector<std::vector<size_t>> & = 0;
+};
+class CombinationGenerator : public WiresGenerator {
```

Comment:
A quick docstring about the class purpose for all of the classes here would be great.

Reply:
Added!